modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelect.scala

```diff
@@ -50,22 +56,28 @@ object MajorityPeerSelect {
 
     def filterPeerList(peers: NonEmptyList[Peer]): F[NonEmptyList[Peer]] =
       peers
-        .traverse(snapshotClient.getLatestOrdinal(_))
-        .map {
-          _.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+        .parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.getLatestOrdinal(_))
+        .map { latestOrdinals =>
+          val (majorityOrdinal, _) = latestOrdinals.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+          majorityOrdinal
+        }
+        .flatMap { majorityOrdinal =>
+          peers.parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.get(majorityOrdinal).run(_).attempt)
         }
-        .flatMap {
-          case (majorityOrdinal, _) =>
-            peers.traverse(snapshotClient.get(majorityOrdinal).run(_).flatMap(_.toHashed.map(_.hash)))
+        .flatMap { maybeSnapshots =>
+          MonadThrow[F].fromOption(
+            maybeSnapshots.collect { case Right(snapshot) => snapshot }.toNel,
+            NoPeersToSelect
+          )
         }
-        .map(_.zip(peers))
-        .map(_.groupMap { case (hash, _) => hash } { case (_, ps) => ps })
-        .map(_.maxBy { case (_, peers) => peers.size })
-        .map { case (_, peerCandidates) => peerCandidates }
+        .flatMap(_.traverse(_.toHashed.map(_.hash)))
+        .map(_.zip(peers).groupMap { case (hash, _) => hash } { case (_, ps) => ps })
+        .map(_.values.toList.sortWith(_.size > _.size))
```

Comment:
`sortBy(_.size)`

Reply:
@sdavidp so it should be `sortBy(-_.size)` then

Reply:
@marcinwadon Why is that preferred over `sortWith` ?

Reply:
also, I see no `sortWith` in the codebase but I see 8 usages of `sortBy`

Reply:
Changed.